src/containers/LinkedCells.h

```diff
@@ -216,6 +221,38 @@ class LinkedCells : public ParticleContainer<Particle, ParticleCell> {
    */
   CellBlock3D<ParticleCell> _cellBlock;
   // ThreeDimensionalCellHandler
+
+  /**
+   * Iterate over all cells and load the data in the SoAs
+   * @tparam ParticleFunctor
+   * @param functor
+   */
+  template <class ParticleFunctor>
+  void fillSoAs(ParticleFunctor &functor) {
```

Comment:
1. seems like you have chosen a different name for these functions compared to the verlet-list-vectorization branch.
I had: loadSoA and extractSoA
to make it more consistent with the function names of the Functor. about the `s` I dont really care, so you can leave it as is and I will change it in the Verletlists.
2. the `&` should probably be a `*`, otherwise the `functor->` looks awkward. you are also calling this function with a `Functor *`. so in the end, you have here: `Functor*&` so a functor pointer by reference

Reply:
Let's stick to the plural since we treat all cells ;)